Document ADR 0008 about unrecognized fields

MVrachev opened the pull request:
Please fill in the fields below to submit a pull request.  The more information
that is provided, the better.

Fixes #1266 

**Description of the changes being introduced by the pull request**:
Even though, this ADR documents something already implied in the TUF
spec in [document formats](https://theupdateframework.github.io/specification/latest/#document-formats)
it seems better to document this decision clearly so that it could be
referenced and give an explanation why someone can load a metadata file
with additional unrecognized fields.

Signed-off-by: Martin Vrachev <mvrachev@vmware.com>

**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

Commit:
Document ADR 0008 about unrecognized fields

Even though, this ADR documents something already implied in the TUF
spec in [document formats](https://theupdateframework.github.io/specification/latest/#document-formats)
it seems better to document this decision clearly so that it could be
referenced and give an explanation why someone can load a metadata file
with additional unrecognized fields.

Signed-off-by: Martin Vrachev <mvrachev@vmware.com>

mnm678 reviewed (approved):
This looks great, thanks @MVrachev! This should help us move forward with some of the TAPs

joshuagl commented on docs/adr/0008-accept-unrecognised-fields.md:
```diff
@@ -0,0 +1,44 @@
+This leads to the side effect that if you read a metadata file with unrecognized
+fields and immediately write it back to the disk, this file will be modified.
+
+Furthermore, some TAPs ([6](https://github.com/theupdateframework/taps/blob/master/tap6.md), [10](https://github.com/theupdateframework/taps/blob/master/tap10.md), [14](https://github.com/theupdateframework/taps/blob/master/tap14.md), [15](https://github.com/theupdateframework/taps/blob/master/tap15.md), and [16](https://github.com/theupdateframework/taps/blob/master/tap16.md)) are relying on tha ...
```
Nit: this is a long line, can we wrap it?

joshuagl commented on docs/adr/0008-accept-unrecognised-fields.md:
```diff
@@ -0,0 +1,44 @@
+
+## Decision Drivers
+- The TUF specification implies support for custom attribute-value fields
+See [Document formats](https://theupdateframework.github.io/specification/latest/#document-formats)
```
```suggestion
- The TUF specification implies support for custom attribute-value fields, see [Document formats](https://theupdateframework.github.io/specification/latest/#document-formats)
```

joshuagl commented on docs/adr/0008-accept-unrecognised-fields.md:
```diff
@@ -0,0 +1,44 @@
+intermediate operations:
+1. read the metadata file
+2. write the metadata file back to the disk
+the checksum (the content) of the file must not be changed.
```
this Markdown is rendered such that "the checksum (the content) of the file must not be changed" appears on the same line as "write the metadata file back to the disk"

joshuagl reviewed (approved):
Thanks for capturing this @MVrachev. I made a few minor comments, otherwise this looks good.

MVrachev commented:
@joshuagl I addressed your comments and amend the commit so it can be ready to be merged.

PS: I had to force-push a couple of times to fix some formatting issues.

joshuagl commented on docs/adr/0008-accept-unrecognised-fields.md:
```diff
@@ -0,0 +1,52 @@
+don't recognize the field.
+
+## Decision Drivers
+- The TUF specification implies support for custom attribute-value fields,
```
```suggestion
- The TUF specification implies support for unrecognised attribute-value fields,
```

joshuagl commented on docs/adr/0008-accept-unrecognised-fields.md:
```diff
@@ -0,0 +1,52 @@
+2. write the metadata file back to the disk
+
+then, the checksum (the content) of the file must not be changed.
+- Flexibility to add new fields in the spec without adding breaking changes.
```
I think this is and the second bullet (backwards compatibility) are the same? This is just more descriptive of what we mean by that term. Let's remove the 2nd bullet?

MVrachev commented:
I am not sure why, but we have a CI failure again on Windows
https://github.com/theupdateframework/tuf/pull/1343/checks?check_run_id=2342301986
@jku does this look familiar to other Windows CI failures?

jku commented:
Not at all, this is tox calling pip and pip failing to download "typed-ast" in some fantastic way...

jku commented:
For reference: the CI failure seems like a pypi problem (made worse by our decision to use pre-release build tooling so downloading more often). Disappeared on re-run, I'm not filing any issues on that. Merging this.
